src/uu/tr/src/operation.rs

```diff
@@ -0,0 +1,529 @@
+            (
+                l,
+                Sequence::Char(std::char::from_u32(a.to_digit(8).unwrap()).unwrap()),
+            )
+        })
+    }
+
+    fn parse_2_octal(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("\\"), one_of("01234567"), one_of("01234567")))(input).map(|(l, (_, a, b))| {
+            (
+                l,
+                Sequence::Char(
+                    std::char::from_u32(a.to_digit(8).unwrap() * 8 + b.to_digit(8).unwrap())
+                        .unwrap(),
+                ),
+            )
+        })
+    }
+
+    fn parse_3_octal(input: &str) -> IResult<&str, Sequence> {
+        tuple((
+            tag("\\"),
+            one_of("01234567"),
+            one_of("01234567"),
+            one_of("01234567"),
+        ))(input)
+        .map(|(l, (_, a, b, c))| {
+            (
+                l,
+                Sequence::Char(
+                    // SAFETY: All the values from \000 to \777 is valid based on a test below...
+                    std::char::from_u32(
+                        a.to_digit(8).unwrap() * 8 * 8
+                            + b.to_digit(8).unwrap() * 8
+                            + c.to_digit(8).unwrap(),
+                    )
+                    .unwrap(),
+                ),
+            )
+        })
+    }
+
+    fn parse_backslash(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("\\"), tag("\\")))(input).map(|(l, _)| (l, Sequence::Char('\\')))
+    }
+
+    fn parse_audible_bel(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("\\"), tag("a")))(input).map(|(l, _)| (l, Sequence::Char('\u{0007}')))
+    }
+
+    fn parse_backspace(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("\\"), tag("b")))(input).map(|(l, _)| (l, Sequence::Char('\u{0008}')))
+    }
+
+    fn parse_form_feed(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("\\"), tag("f")))(input).map(|(l, _)| (l, Sequence::Char('\u{000C}')))
+    }
+
+    fn parse_newline(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("\\"), tag("n")))(input).map(|(l, _)| (l, Sequence::Char('\u{000A}')))
+    }
+
+    fn parse_return(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("\\"), tag("r")))(input).map(|(l, _)| (l, Sequence::Char('\u{000D}')))
+    }
+
+    fn parse_horizontal_tab(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("\\"), tag("t")))(input).map(|(l, _)| (l, Sequence::Char('\u{0009}')))
+    }
+
+    fn parse_vertical_tab(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("\\"), tag("v")))(input).map(|(l, _)| (l, Sequence::Char('\u{000B}')))
+    }
+
+    fn parse_char_range(input: &str) -> IResult<&str, Sequence> {
+        separated_pair(take(1usize), tag("-"), take(1usize))(input).map(|(l, (a, b))| {
+            (l, {
+                let (start, end) = (
+                    u32::from(a.chars().next().unwrap()),
+                    u32::from(b.chars().next().unwrap()),
```

Comment:
Why do you cast them to `u32`?

Reply:
I am not sure where this was but it shouldn't be relevant anymore since the past few commits.